docs/verifiability_doc.md

```diff
@@ -0,0 +1,172 @@
+# vote verifiability
+
+## Introduction
+
+Verifiability is an important property to allows voters to check their vote has been cast unaltered, and that it has been registered correctly in the electronic ballot box.
+
+The current d-voting (version num) didn't have this design yet. The current encrypted ballot logic is shown as follows:
+
+```mermaid
+ sequenceDiagram
+    autonumber
+    participant User
+    participant Backend
+    participant NodeX
+    participant NodeY
+    User ->>+ NodeX: GET election info
+    NodeX ->>- User: Return election info.
+    User ->>+ Backend: POST /api/evoting/elections/<electionId>
+    Note over User: data: {"Ballot": ...}
+    Note over User: encrypt ballot via Elgamal encryption using electionPubKey
+    Note over User, Backend: data = encrypted ballot
+    Note over Backend: check role and sign payload.
+    Note over Backend: add userID inside payload.
+    Note over Backend: sign = kyber.sign.schnorr.sign(edCurve, scalar, hash);
+    Backend ->>+ NodeX: POST /evoting/elections/
+    Note over Backend, NodeX: data: {"Payload": dataStrB64, "Signature": ""}
+    Note over NodeX: verify and execute, then boardcast 
+    NodeX ->> NodeY: boardcase via gRPC
+    NodeX ->>- Backend: 200 OK text/plain
+    Backend ->>- User: 200 OK text/plain
+```
+
+As the picture show, the Frontend will encrypt the ballot using Elgamal encryption which has a nondeterministic result and then sends it to the Backend to verify and sign. After that, the backend will send the encrypted + signed ballot to the blockchain node to put it on the chain. However, since the encryption is nondeterministic thus the user will not able to verify their casted ballot stored in the node.

```

Comment:
```suggestion
As the picture shows, the frontend encrypts the ballot using Elgamal encryption which has a nondeterministic result and then sends it to the backend to verify and sign. After that, the backend sends the encrypted + signed ballot to the blockchain node to put it on the chain. However, since the encryption is nondeterministic the user will not able to verify their casted ballot stored in the node.

```